packages/amplify-graphql-auth-transformer/src/__tests__/resolvers/__snapshots__/subscriptions.test.ts.snap

```diff
@@ -0,0 +1,395 @@
+## [End] Authorization Steps. **"
+`;
+
+exports[`subscriptions multi-auth renders for iam + userPools 1`] = `
+"## [Start] Authorization Steps. **
+$util.qr($ctx.stash.put(\\"hasAuth\\", true))
+#set( $isAuthorized = false )
+#if( $util.authType() == \\"IAM Authorization\\" )
+  #if( !$isAuthorized )
+    #if( $ctx.identity.userArn == $ctx.stash.undefined )
+      #set( $isAuthorized = true )
+    #end
+  #end
+  #if( !$isAuthorized )
+    #if( $ctx.identity.userArn == $ctx.stash.undefined )
+      #set( $isAuthorized = true )
+    #end
+  #end
+#end
+#if( $util.authType() == \\"User Pool Authorization\\" )
+  #set( $hasValidOwnerArgument = false )
+  #if( !$isAuthorized )
+    #set( $staticGroupRoles = [{\\"claim\\":\\"cognito:groups\\",\\"entity\\":\\"Admin\\"}] )
+    #foreach( $groupRole in $staticGroupRoles )
+      #set( $groupsInToken = $util.defaultIfNull($ctx.identity.claims.get($groupRole.claim), []) )
+      #if( $groupsInToken.contains($groupRole.entity) )
+        #set( $isAuthorized = true )
+        #break
+      #end
+    #end
+  #end
+  #set( $authRuntimeFilter = [] )
+  #set( $authOwnerRuntimeFilter = [] )
+  #set( $authGroupRuntimeFilter = [] )
+  #set( $ownerClaim0 = $util.defaultIfNull($ctx.identity.claims.get(\\"username\\"), $util.defaultIfNull($ctx.identity.claims.get(\\"cognito:username\\"), null)) )
+  #if( !$util.isNull($ownerClaim0) )
+
+    #set( $ownerClaimsList0 = [] )
+    $util.qr($authOwnerRuntimeFilter.add({ \\"owner\\": { \\"eq\\": $ownerClaim0 } }))
+    #set( $ownerEntity0 = $util.defaultIfNull($ctx.args.owner, null) )
+    #if( !$isAuthorized && !$util.isNullOrEmpty($ownerEntity0) )
+      #if( $ownerEntity0 == $ownerClaim0 || $ownerClaimsList0.contains($ownerEntity0) )
+        #set( $isAuthorized = true )
+        #set( $hasValidOwnerArgument = true )
+      #else
+        $util.unauthorized()
+      #end
+    #end
+  #end
+  #set( $groupClaim1 = $util.defaultIfNull($ctx.identity.claims.get(\\"cognito:groups\\"), []) )
+  #if( $util.isString($groupClaim1) )
+    #if( $util.isList($util.parseJson($groupClaim1)) )
+      #set( $groupClaim1 = $util.parseJson($groupClaim1) )
+    #else
+      #set( $groupClaim1 = [$groupClaim1] )
+    #end
+  #end
+  $util.qr($authGroupRuntimeFilter.add({ \\"group\\": { \\"in\\": $groupClaim1 } }))
+  ## Apply dynamic roles auth if not previously authorized by static groups and owner argument **
+  #if( $authOwnerRuntimeFilter.size() > 0 )
+    $util.qr($authRuntimeFilter.addAll($authOwnerRuntimeFilter))
+  #end
+  #if( $authGroupRuntimeFilter.size() > 0 )
+    $util.qr($authRuntimeFilter.addAll($authGroupRuntimeFilter))
+  #end
+  #set( $isOwnerAuthAuthorizedAndNoOtherFilters = $hasValidOwnerArgument && $authRuntimeFilter.size() == 1 && $ctx.args.filter.size() == 0 )
+  #if( !$isOwnerAuthAuthorizedAndNoOtherFilters && (!$isAuthorized || $hasValidOwnerArgument) && $authRuntimeFilter.size() > 0 )
+    #if( $util.isNullOrEmpty($ctx.args.filter) )
```

Comment:
I feel like we can simplify the above 3 lines, given the change suggested above.

Reply:
It's refactored in the latest rev, but uses the same boolean logic (just pulled out into variable names)